app/Http/Controllers/BlogController.php

```diff
@@ -2,47 +2,278 @@
+    {
+        
+        $this->validate($request, [
+            'title' => 'required|min:5',
+            'content' => 'required|min:10'
+        ]);
+
+        if (!$user = JWTAuth::parseToken()->authenticate()) {
+          return response()->json(['message' => 'User not found'], 404);
+        }
+
+        $user_id = $user->id;
+        $currentUser = User::where("id", $user_id)->first();
+    
+        $post = new Post([
+          'title' => $request->input('title'),
+          'content' => $request->input('content'),
+          "author" => $currentUser->username
+        ]);
+        
+        $currentUser->blogPosts()->save($post);
+
+        return response()->json(['success' => true, 'The blog post has been created']);
+    }
+
+    /**
+     * Edit a created blog post
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function editPost(Request $request, $postId)
+    {
+        
+        $this->validate($request, [
+            'title' => 'required|min:5',
+            'content' => 'required|min:10'
+        ]);
+
+        if (!$user = JWTAuth::parseToken()->authenticate()) {
+          return response()->json(['message' => 'User not found'], 404);
+        }
+
+        $user_id = $user->id;
+
+        $post = new Post();
+        $editedPost = $post->where([
+            ['user_id', '=', $user_id],
+            ['id', '=', $postId],
+        ])->update([
+            'title' => $request->input('title'),
+            'content' => $request->input('content')
+        ]);
+
+        return response()->json(['success' => true,
+            'message' => 'The blog post has been edited'
+        ]);
+    }
+
+    /**
+     * Delete a created blog post
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function deletePost(Request $request, $postId)
+    {
+
+        if (!$user = JWTAuth::parseToken()->authenticate()) {
+          return response()->json(['message' => 'User not found'], 404);
+        }
+
+        $user_id = $user->id;
+
+        $post = new Post();
+        $postToDelete = $post->where([
+            ['user_id', '=', $user_id],
+            ['id', '=', $postId],
+        ])->delete();
+
+        return response()->json(['success' => true,
+            'message' => 'The blog post has been deleted'
+        ]);
+    }
+
+
+    /**
+     * Like a post
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function likePost(Request $request, $postId)
+    {
+
+        if (!$user = JWTAuth::parseToken()->authenticate()) {
+            return response()->json(['message' => 'User not found'], 404);
+        }
+    
+        $user_id = $user->id; 
+        
+        $like = new Like();
+
+        $likedPost = $like->where([
+            ['user_id', '=', $user_id],
+            ['post_id', '=', $postId],
+        ])->first();
+
+        if (is_null($likedPost) == false) {
```

Comment:
In future, `$likedPost` signature might not be null but any other falsy values so it is safer that you check for other falsy data types as well.

```
if(!$likedPost){
...
}
```